tailcfg/tailcfg.go

```diff
@@ -1897,39 +1892,88 @@ type TKAInitFinishRequest struct {
 // key authority.
 type TKAInitFinishResponse struct{}
 
-// TKAMapRequest describes request parameters relating to the tailnet key
-// authority instance on this node. This information is transmitted as
-// part of the MapRequest.
-type TKAMapRequest struct {
-	// Head is the AUMHash of the latest authority update message committed
-	// by this node.
-	Head string // tka.AUMHash.String
-}
-
-// TKAMapResponse encodes information for the tailnet key authority
-// instance on the node. This information is transmitted as
-// part of the MapResponse.
-//
-// If there are no updates to be transmitted (in other words, if both
-// control and the node have the same head hash), len(Updates) == 0 and
-// WantSync is false.
-//
-// If control has updates that build off the head hash reported by the
-// node, they are simply transmitted in Updates (avoiding the more
-// expensive synchronization process).
-//
-// In all other cases, WantSync is set to true, and the node is expected
-// to reach out to control separately to synchronize.
-type TKAMapResponse struct {
-	// Updates is any AUMs that control believes the node should apply.
-	Updates []tkatype.MarshaledAUM `json:",omitempty"`
-
-	// WantSync is set by control to request the node complete AUM
-	// synchronization.
+// TKAInfo encodes controls' view of tailnet key authority (TKA) state.
+// This information is transmitted as part of the MapResponse.
+type TKAInfo struct {
+	// Head describes the hash of the latest AUM applied to the authority.
+	// Head is encoded as tka.AUMHash.MarshalText()
 	//
-	// TODO(tom): Implement AUM synchronization, probably as noise endpoints
+	// If the Head state differs to that known locally, the node should perform
+	// synchronization via a separate RPC.
+	//
+	// TODO(tom): Implement AUM synchronization as noise endpoints
 	// /machine/tka/sync/offer & /machine/tka/sync/send.
-	WantSync bool `json:",omitempty"`
+	Head string `json:",omitempty"`
+
+	// Enabled describes whether the tailnet key authority should be enabled.
+	//  - False: TKA should be disabled. If it is enabled locally, the node
+	//    should reach out to fetch a disablement secret. If the disablement
+	//    secret verifies, then the node should then disable TKA locally.
+	//  - True: TKA should be enabled. If it is not, the node should reach
+	//    out via separate RPC to fetch the genesis AUM and initialize.
+	//
+	// TODO(tom): Implement /machine/tka/boostrap as a noise endpoint, to
+	// communicate the genesis AUM & any disablement secrets.
+	Enabled bool `json:",omitempty"`
```

Comment:
probably don't need omitempty here. the outer one's already omit empty so when we're looking at this output with TS_DEBUG_MAP=1 it'll be useful to see an explicit "Enabled: false" fly by on stderr.